pkg/ruler/base/ruler.go

```diff
@@ -122,24 +123,19 @@ type Config struct {
 }
 
 // Validate config and returns error on failure
-func (cfg *Config) Validate(log log.Logger) error {
-	// support for deprecated sharding configuration
-	level.Warn(log).Log("msg", fmt.Sprintf("use of deprecated ruler sharding strategy %s, instead use enable_shuffle_sharding", util.ShardingStrategyShuffle))
+func (cfg *Config) Validate(_ log.Logger) error {
 
-	if cfg.ShardingStrategy == util.ShardingStrategyShuffle {
-		cfg.EnableShuffleSharding = true
-
-		// TODO(v3.0): replace with util.ShardingStrategyByRule
+	if cfg.ShardingStrategy == "" {
```

Comment:
I think we can remove this clause. We already set the defaults in `RegisterFlags` -- let's just do validation here but not mutate the config if we can avoid it.

Reply:
I had these here because of the tests (which don't initialise the config the way that `RegisterFlags` does).
ideally we should have a way to "instantiate" these configs without requiring the flags to do it for us, so I'll leave this as is for the interim.